src/http-client/http-client.ts

```diff
@@ -0,0 +1,69 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Client } from "../client";
+import { QueryRequest } from "../wire-protocol";
+
+/**
+ * An object representing an http request.
+ * The {@link Client} provides this to the {@link HTTPClient} implementation.
+ */
+export type HTTPRequest = {
+  /** The encoded Fauna query to send */
+  data: QueryRequest;
+
+  /** Headers in object format */
+  headers: Record<string, string | undefined>;
+
+  /**
+   * Time in milliseconds the client will keep an HTTP2 session open after all
+   * requests are completed. Only necessary for HTTP2 implementations.
+   */
+  http2_sessions_idle_ms: number;
+
+  /** HTTP method to use */
+  method: "POST";
+
+  /** URL for the request */
+  url: string;
+
+  /**
+   * The timeout of each query, in milliseconds. This controls the maximum amount of
+   * time Fauna will execute your query before marking it failed.
+   */
+  client_timeout_ms?: number;
```

Comment:
this is confusing.

This is not really controlling a query timeout - its the HTTP client timeout.

Let's set this as the buffer as well.

We can pass down a query_timeout_ms in this type as well.

That way the client timeout can use the simple query_timeout_ms + client_timeout_ms math.

If query_timeout is undefined then we know to use core's default as the query_timeout_ms basis.

Reply:
alternative is to infer the client timeout from defaults in the client.ts and never pass undefined here. and have the buffer be a client only concern. that might be cleaner